src/Chunk.cpp

```diff
@@ -1885,7 +1885,13 @@ bool cChunk::AddClient(cClientHandle * a_Client)
 
 void cChunk::RemoveClient(cClientHandle * a_Client)
 {
-	m_LoadedByClient.erase(std::remove(m_LoadedByClient.begin(), m_LoadedByClient.end(), a_Client));
+	auto itr = std::remove(m_LoadedByClient.begin(), m_LoadedByClient.end(), a_Client);
+	// We should always remove at most one client.
+	ASSERT(std::distance(itr, m_LoadedByClient.end()) <= 1);
+	if (itr != m_LoadedByClient.end())
```

Comment:
I could, but by putting in the if, I'm reminding myself that them being equal is an issue.

Personally, given it has to be zero or one, forcing it rather than a range, ensures that we have better handling for unexpected values.